draft-ietf-jsonpath-base.md

```diff
@@ -1068,22 +1068,22 @@ Note that there is no "bald" descendant selector `..`.
 #### Semantics
 {: unnumbered}
 
-The `descendant-selector` selects certain descendants of a node:
+A `descendant-selector` selects certain descendants of a node:
 
 * the `..<name>` form (and the `..[<index>]` form where `<index>` is a `quoted-member-name`) selects those descendants that are member values of an object with the given member name.
 * the `..[<index>]` form, where `<index>` is an `element-index`, selects those descendants that are array elements with the given index.
 * the `..[<slice-index>]` form selects those descendants that are array elements selected by the given slice.
 * the `..[<filter>]` form selects those descendants that are array elements or object values selected by the given filter.
 * the `..[*]` and `..*` forms select all the descendants.
 
-The resultant nodelist is ordered as if:
+An _array preorder_ of the descendants of a node is a sequence of all the descendants in which:
 
-* nodes are visited before their children, and
-* nodes of an array are visited in array order.
+* nodes of an array appear in array order, and
+* nodes appear before their children.
```

Comment:
hmm, this doesn't actually say depth first.

..* on
[1, 2, 3, [4, 5], 6]

could be shown as [1, 2, 3, [4, 5], 6, 4, 5] and satisfy the two rules.

Reply:
Good point. I'll address that shortly.